### ISSUE TITLE
Simplify to find+erase in native pipeline

### ISSUE THREAD

**Issue by jdgarciauc3m** [arcosuc3m/grppi#229]
In native *pipeline* implementation there is a common pattern:

~~~c++
      for (auto it=elements.begin(); it!=elements.end(); it++) {
        if(it->second == current) {
          consume(*it->first);
          elements.erase(it);
          current++;
          break;
        }
      }
~~~

The explicit loop may be avoided by combining `find_if` and `erase`:
~~~c++
auto it = find_if(elements.begin(), elements.end(), 
  [](auto x) { return x.second== current; });
if(it != elements.end()){
  consume(*it->first);
  elements.erase(it);
  current = it->second+1;
}
~~~

**Comment by jdgarciauc3m** [comment]
@Drioastorga also suggested

~~~c++
auto it = find_if(elements.begin(), elements.end(), current == it->second);
if(it != elements.end()){
  consume(it->first.value());
  elements.erase(it);
  current++;
}
~~~

But I am not sure, if this provides the same semantics.

**Comment by jdgarciauc3m** [comment]
Solved by pr #345
